### PR TITLE
migrate to oclif

### LINKED ISSUE #90: Migrate to oclif
- [ ] switch back to mocha via codemod (if needed) #91 
- [ ] run a codemod for oclif testing #92 
- [ ] migrate commands to oclif #93

### LINKED ISSUE #91: Switch back to mocha
oclif has mocha-based test helpers and it should be a pretty straightforward switch

### LINKED ISSUE #92: migrate tests to be use oclif helpers

### LINKED ISSUE #93: reimplement commands via oclif

### PR DESCRIPTION
**Pull request by ianwremmel**
Fixes #93
Fixes #92
Fixes #91
Fixes #90

## Blockers
- [x] put back global/local support
- [x] fix `--silent`
- [x] support `--package`, `--package-name`, and `--packageName`
- [x] update README
- [x]  ensure there's at least one breaking change commit
- [x] wireup new publication process

## Future Tasks
- #103 consider deprecating `.clarkrc` in favor of oclif's config tool
- #102 remove casted reduces (`.reduce<>`)
- #101 output prints too  many linebreaks
- #100 `run` no longer prints `.clarkrc` scripts
- #99 superclass `exec`, `hoist`, `run` and generalize the kebab/camelcaseLogic
- #98 restore nyc

### CODE REVIEW

File: src/lib/packages.ts [hunk 1]
```diff
@@ -68,10 +69,10 @@ export namespace apply {
    * Logger definitions
    */
   export interface Options {
-    before: (packages: string[]) => string;
-    beforeEach: (packageName: string) => string;
-    afterEach: (packageName: string, error?: Error) => string;
-    after: (packages: string[], errors: Error[]) => string;
+    before(packages: string[]): string;
```
Comment (kourge) [inline src/lib/packages.ts]: Although the two are equivalent, the method form usually implies that function works in conjunction with a `this` context, whereas the function property form implies that the function works anywhere.

Comment (ianwremmel) [reply src/lib/packages.ts]: noted. I'm using the oclif lint rules without modification for the moment. it insisted on this form.

File: src/lib/packages.ts [hunk 2]
```diff
@@ -453,10 +465,11 @@ export async function list(): Promise<string[]> {
  * @param pattern
  */
 async function listPackagesInGlob(pattern: string): Promise<void> {
-  debug(f`Listing packages in ${pattern}`);
+  const cwd = await findProjectRoot();
+  debug(f`Listing packages in ${cwd} matchign ${pattern}`);
```
Comment (kourge) [inline src/lib/packages.ts]: typo: "matching"

**Pull request merged**